### PR TITLE
win,libuv: support multiple threads

### LINKED ISSUE #32087: repeated threads loop hangs on windows
Calling this function on windows:
```
function f()
    nt = Threads.nthreads()
    for k in 1:5
        v = zeros(32 * nt)
        Threads.@threads for n in 1:nt
            for i in 1:100000000
                v[32 * n] += i
            end
        end
        @show k, v[32 .* (1:nt)]
    end
end
```
with JULIA_NUM_THREADS > 1 ends up hanging, using 100% of one CPU. It sometimes gets through some iterations of the outer loop first. Can anybody reproduce this? Works fine on linux.

### PR DESCRIPTION
**Pull request by vtjnash**
turns off the "do not work" flag

fix #32087

### CODE REVIEW

File: deps/libuv.version [hunk 1]
```diff
@@ -1,2 +1,2 @@
 LIBUV_BRANCH=julia-uv2-1.24.0
-LIBUV_SHA1=2348256acf5759a544e5ca7935f638d2bc091d60
+LIBUV_SHA1=26dbe5672c33fc885462c509fe2a9b36f35866fd
```
Comment (vtjnash) [inline deps/libuv.version]: https://github.com/libuv/libuv/compare/2348256acf5759a544e5ca7935f638d2bc091d60...26dbe5672c33fc885462c509fe2a9b36f35866fd

**Review by JeffBezanson — APPROVED** [review]

**Comment by Keno** [comment]
As I commented on the relevant commit somewhere. I don't think this fix is sufficient in the presence of more threads than the number of cores in the machine. The problem is that threads retain their completion port association, but we migrate the completion port to a different thread. Ideally we'd have a way to drop that association and keep this value as is.

**Comment by JeffBezanson** [comment]
That makes sense, but in the meantime this is a much-needed improvement. It makes the difference between simple threaded code locking up every time vs. generally working.

**Comment by vtjnash** [comment]
That's something to consider in the future, although hopefully less of a concern, since any blocking call to the kernel should enable releasing a new thread from the IOCP. If we do continue with the current IO design unchanged, there is a way that this should be addressable per the MSDN documentation:

> Although any number of threads can call GetQueuedCompletionStatus for a specified I/O completion port, when a specified thread calls GetQueuedCompletionStatus the first time, it becomes associated with the specified I/O completion port until one of three things occurs: The thread exits, **specifies a different I/O completion port**, or closes the I/O completion port. In other words, a single thread can be associated with, at most, one I/O completion port.

**Comment by Keno** [comment]
Can we just set that value to something like typemax? Are there any resources allocated proportional to that value? If not, setting it really high is probably a solution, since we can't allocate that many threads anyway.

**Pull request merged**